akka-projection-slick/src/main/scala/akka/projection/slick/SlickProjection.scala

```diff
@@ -23,19 +24,30 @@ object SlickProjection {
   def exactlyOnce[Offset, Envelope, P <: JdbcProfile: ClassTag](
       projectionId: ProjectionId,
       sourceProvider: SourceProvider[Offset, Envelope],
-      databaseConfig: DatabaseConfig[P])(eventHandler: Envelope => DBIO[Done]): Projection[Envelope] =
+      databaseConfig: DatabaseConfig[P],
+      eventHandler: SlickEventHandler[Envelope]): Projection[Envelope] =
```

Comment:
I remove the functional variant because I want to have less things to fix while moving things around. 

I will re-add for comparison (tomorrow).

Reply:
Maybe we could support both by making the `EventHandler` extend `Function1`. A user could use either functional or type implementation usage and internally we could add default behaviour for other things in your `EventHandler` type when called functionally.

i.e.

```scala
type EventHandler = (Envelope) => DBIO[Done]

trait SlickEventHandler extends EventHandler {
  def apply(env: Envelope): DBIO[Done] = handleEvent(env)
  def handleEvent(envelope: Envelope): DBIO[Done]
  ...
}

// API

def atLeastOnce(eventHandler: EventHandler) = eventHandler match {
  case e: EventHandler => // functional case
  case s: SlickEventHandler => // type implementation case
}

// functional usage
atLeastOnce((env) => something)

// type impl.
atLeastOnce(new SlickEventHandler { 
  def handleEvent(env: Envelope): DBIO[Done] = ???
})
```

Reply:
I would like to experiment with this idea on another iteration. For the moment I added something as suggested here https://github.com/akka/akka-projection/pull/82#discussion_r418027000